### PR TITLE
feat(SWH): add time to reset if X-RateLimit-Limit reached for SWH agent

### LINKED ISSUE #1647: Update info about REST rate throttling at SW Heritage agent
# Description

When running the SW Heritage agent, after 120 files, apparently the REST rate throtteling kicks in. So there will be a throttling and the user has to wait until the next batch of files can be processed.

It would be good to inform the user about the current throttling status in the agent log on the agent geeky scan details page.

#### Screenshots

Screenshot 2020-03-05 at 09 19 55

### PR DESCRIPTION
**Pull request by shaheemazmalmmd**
closes #1647 
## Description

Add time to reset to agent job log.

## How to test

* Scan SWH agent with a package more than 120 files and check the job log.

### CODE REVIEW

File: src/softwareHeritage/agent/softwareHeritageAgent.php [hunk 1]
```diff
@@ -129,6 +129,7 @@ function processEachPfileForSWH($pfileDetail, $agentId, $maxTime)
       } else {
         sleep($timeToReset);
       }
+      print "INFO :Software Heritage X-RateLimit-Limit reached. Next slot unlocks in ".gmdate("H:i:s", $timeToReset)." ";
```
Comment (GMishx) [inline src/softwareHeritage/agent/softwareHeritageAgent.php]: IMO, the message should be printed before going to sleep.

Comment (shaheemazmalmmd) [reply src/softwareHeritage/agent/softwareHeritageAgent.php]: Does it really make a difference ??

Comment (GMishx) [reply src/softwareHeritage/agent/softwareHeritageAgent.php]: The idea of the message is to inform user why agent is not processing more files.

Also, let's say if the rate limit will reset in 30 seconds and you've already paused for 30 seconds and then inform user you are going to pause for 30 seconds, the math does not add up.

File: src/softwareHeritage/agent/softwareHeritageAgent.php [hunk 2]
```diff
@@ -124,6 +124,7 @@ function processEachPfileForSWH($pfileDetail, $agentId, $maxTime)
     if (SoftwareHeritageDao::SWH_RATELIMIT_EXCEED == $currentStatus) {
       $this->heartbeat(0); //Fake heartbeat to keep the agent alive.
       $timeToReset = $currentResult - time();
+      print "INFO :Software Heritage X-RateLimit-Limit reached. Next slot unlocks in ".gmdate("H:i:s", $timeToReset)." ";
```
Comment (GMishx) [inline src/softwareHeritage/agent/softwareHeritageAgent.php]:
```suggestion
      print "INFO :Software Heritage X-RateLimit-Limit reached. Next slot unlocks in ".gmdate("H:i:s", $timeToReset)."\n";
```

While testing the PR, I saw some unusual logs:

```
"INFO :Software Heritage X-RateLimit-Limit reached. Next slot unlocks in 00:56:05 HEART: 47 1"
"INFO :Software Heritage X-RateLimit-Limit reached. Next slot unlocks in 00:55:40 HEART: 47 1"
"INFO :Software Heritage X-RateLimit-Limit reached. Next slot unlocks in 00:55:09 HEART: 47 1"
"INFO :Software Heritage X-RateLimit-Limit reached. Next slot unlocks in 00:54:37 HEART: 47 1"
```

Potentially the hearbeats to the scheduler are getting lost. Adding the newline to the end of message improved the situation:

```
"INFO :Software Heritage X-RateLimit-Limit reached. Next slot unlocks in 00:50:22"
"INFO :Software Heritage X-RateLimit-Limit reached. Next slot unlocks in 00:49:53"
"INFO :Software Heritage X-RateLimit-Limit reached. Next slot unlocks in 00:49:21"
"INFO :Software Heritage X-RateLimit-Limit reached. Next slot unlocks in 00:48:50"
```

Please have a look @shaheemazmalmmd

**Pull request merged**